specification.md

```diff
@@ -4313,28 +4429,415 @@ transition: IssueRefundToCustomer
+        "when": [
+          { "status": 503 }
+        ],
+        "retry": "my-retry-policy",
+        "then": {
+          "transition": "my-state"
+        }
+      }
+    ]
+  }
+}
+```
+
+</td>
+<td valign="top">
+
+```yaml
+errors:
+  handlers:
+    - name: 'handle-invalid-error'
+      when:
+    	- type: 'invalid'
+    	- status: 404
+    	- status: 403
+      then:
+    	  transition: 'my-state'
+    - name: 'handle-timeout-error'
+      when:
+    	- status: 503
+      retry: 'my-retry-policy'
+      then:
+        transition: 'my-state'
+```
+
+</td>
+</tr>
+</table>
+
+</details>
+
+Error handler definitions specify which errors to handle and how they should be handled within a workflow.
+
+The `name` property specifies the distinct identifier utilized to reference the error handler.
+
+The `when` property defines the specific errors to handle. Allows for handling only specific errors.
+
+The `exceptWhen` property defines the specified errors NOT to handle. Allows for handling all errors, excluding specific ones.
+
+The `retry` property serves to either reference an existing retry policy or define a new one to be employed when handling specified errors within the workflow. If a retry policy is designated, the error source identified by the [error source](#error-source) will undergo retries according to the guidelines outlined in the associated [policy](#retry-definition). In the event that a retry attempt is successful, the workflow seamlessly proceeds as though the error had not transpired. However, if the maximum number of configured retry attempts is exhausted without success, the workflow proceeds to execute the error outcome stipulated by the `then` property.
+
+The `then` property defines caught errors outcome, if any. If not defined, caught errors will be considered as handled, and the execution of the workflow will continue as if the error never occured. Handled errors that are not [rethrown](#error-outcome-definition) do NOT [bubble up](#error-bubbling).
 
 For more information, see the [Workflow Error Handling](#Workflow-Error-Handling) sections.
 
+##### Error Handler Reference
+
+| Parameter | Description | Type | Required |
+| --- | --- | --- | --- |
+| refName | The name of the error handler definition to reference. If set, all other properties are ignored. | string | no |
+| when | References the errors to handle. If null or empty, and if `exceptWhen` is null or empty, all errors are caught. | array of [errorRefs](#error-reference) | no |
+| exceptWhen | References the errors not to handle. If null or empty, and if `when` is null or empty, all errors are caught. | array of [errorRefs](#error-reference) | no |
+| retry | The retry policy to use, if any. If a string, references an existing [retry definition](#retry-definition). | string or [retry definition](#retry-definition) | no |
+| then | Defines the outcome, if any, when handling errors | [outcomeDef](#outcome-definition) | no |
```

Comment:
Inconsistent naming `outcomeDef` vs `error outcome definition` above. And point to two different concepts (#outcome-definition vs #error-outcome-definition).